clients/conditionalaccesspolicy_test.go

```diff
@@ -0,0 +1,199 @@
+package clients_test
+
+import (
+	"fmt"
+	"os"
+	"testing"
+
+	"github.com/manicminer/hamilton/auth"
+	"github.com/manicminer/hamilton/clients"
+	"github.com/manicminer/hamilton/clients/internal"
+	"github.com/manicminer/hamilton/internal/utils"
+	"github.com/manicminer/hamilton/models"
+)
+
+var (
+	enterpriseAppId = os.Getenv("ENTERPRISE_APP_ID") // Enterprise Apps cannot be created using Graph. Therefore, to simplify testing pass the Enterprise App ID via an Env variable.
```

Comment:
Rather than requiring an ID be specified, could we use a service principal for Microsoft published app that's likely to be present in all tenants, e.g. Microsoft Graph? You can look those up by displayName or by appId.

Reply:
Removed. I now pull an `appId` from the `PublishedApis` map. I can't use Graph, as it's not on the allowed list for first party applications in Conditional Access Policies. Instead, I opted for "AzureDevOps".